src/rust/aggregator/src/data/leaderboards.rs

```diff
@@ -0,0 +1,261 @@
+use anyhow::anyhow;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, Duration, Utc};
+use sqlx::{PgConnection, PgPool, Postgres, Transaction};
+
+use super::{Data, DataAggregationError, DataAggregationResult};
+
+pub struct Leaderboards {
+    pool: PgPool,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+impl Leaderboards {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+struct Competition {
+    id: i32,
+    start: DateTime<Utc>,
+    end: DateTime<Utc>,
+    prize: i64,
+    market_id: BigDecimal,
+    frontends_required: Vec<String>,
+}
+
```

Comment:
I think this is something we should handle with ORM in `models.rs`

The aggregator functionality is start to drift away from ORM and I think thinks will be cleaner and more maintainable if we can use diesel and avoid entire pages of wrapped SQL